### PR TITLE
Check True in error response for RPC client

### PR DESCRIPTION
**Pull request by wchen-r7**
Without this check, it causes Pro to behave differently (malfunction). Please see:
https://github.com/rapid7/metasploit-framework/pull/4823/files#r25544830

### CODE REVIEW

**Comment by bturner-r7** [comment]
LGTM :shipit: Let's wait for @shuckins-r7 to review, however.

**Comment by wchen-r7** [comment]
A PR to revert the changes can be found here: https://github.com/rapid7/metasploit-framework/pull/4859

**Comment by todb-r7** [comment]
@wchen-r7, to land this, you'll want to first revert #4859. Currently, this PR is unlandable due to merge conflicts. However, I believe there is at least one other problem with confusing `true` and mere Truthiness in #4823.

You can fix up this branch by reverting 7ab86be first, then applying bdc9f68 on top of that.

File: lib/msf/core/handler/reverse_http.rb [hunk 1]
```diff
@@ -158,25 +158,14 @@ def setup_handler
   end
 
   #
-  # Simply calls stop handler to ensure that things are cool.
```
Comment (wchen-r7) [inline lib/msf/core/handler/reverse_http.rb]: This is from b27c9b9efc505888e0a4190ee7d6cdf759a0ad58 when I tried to git merge upstream-master to update the branch.

**Comment by todb-r7** [comment]
I'm fairly certain a jira ticket exists for this, and @bturner-r7 mentioned @shuckins-r7 for review.

**Comment by todb-r7** [comment]
The fixes to the original PR, #4895, are captured atomically in 5f8c14c9580ab99e6171ef90584bd791505e66fc and addresses my concerns about some other fixes, btw.

**Comment by shuckins-r7** [comment]
Thanks for the updates @wchen-r7, looks good, merging shortly.

**Pull request merged**